staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -245,6 +221,18 @@ func (r *withRetry) Before(ctx context.Context, request *Request) error {
 		return nil
 	}
 
+	// At this point we've made atleast one attempt, post which the response
+	// body should have been fully read and closed in order for it to be safe
+	// to reset the request body before we reconnect, in order for us to reuse
+	// the same TCP connection.
+	if seeker, ok := request.body.(io.Seeker); ok && request.body != nil {
+		if _, err := seeker.Seek(0, io.SeekStart); err != nil {
+			err = fmt.Errorf("failed to reset request body for %T - %w", request, err)
```

Comment:
See https://github.com/kubernetes/kubernetes/pull/109050/files#r838594245 but the fmt error should probably be:

    fmt.Errorf("failed to reset the request body while retrying a request: %v", err)

Reply:
@smarterclayton I've addressed this review, thank you! PTAL :)